### PR TITLE
add: tests for launchBroker

### PR DESCRIPTION
**Pull request by elit-altum**
added mock function for testing the launch of the broker as a daemon

### CODE REVIEW

**Comment by elit-altum** [comment]
@coditva please review!

File: dashboard/index.js [hunk 1]
```diff
@@ -1,23 +1,38 @@
...
+            {
+                detached: true,
+                // runs the child process independently of parent
+                stdio: ['ignore', 'ignore', 'ignore', 'ipc'],
+                // disconnect all pipes except for ipc
+                // ['stdin', 'stdout', 'stderr', 'ipc']
+            }
+        );
+    } else {
+        brokerProcess = mocks.mockSpawn(
```
Comment (coditva) [inline dashboard/index.js:27]: This is not how we use mocking. You should stub the function that is external in the test itself. The source should never know about the tests.

Ref:
1. Mocking in newman: https://github.com/postmanlabs/newman/blob/develop/test/unit/cli.test.js
2. https://sinonjs.org/

File: dashboard/index.js [hunk 2]
```diff
@@ -28,9 +43,13 @@ const launchBroker = (port) => {
         brokerProcess.unref();
         brokerProcess.disconnect();
 
-        // clean exit the process
-        process.exit(0);
+        if (!isTest) {
+            // clean exit the process
+            process.exit(0);
```
Comment (coditva) [inline dashboard/index.js:48]: Instead of adding a check for `isTest`, you can stub the `process.exit` and assert that it was called.

**Review by coditva — CHANGES_REQUESTED** [review]

File: dashboard/index.js (hunk 1 above)
Comment (elit-altum) [reply dashboard/index.js:27]: @coditva Thank you for this! I have implemented sinonjs for stubbing at https://github.com/elit-altum/newman-dashboard/pull/7 could we move this discussion there?

**Pull request closed without merging**